app/controllers/records_controller.rb

```diff
@@ -1,7 +1,11 @@
 class RecordsController < ApplicationController
   include RecordsControllerBehavior
 
-  private
+  protected
+
+  def set_attributes
+    resource.attributes = AttributeURIConverter.new(collect_form_attributes).convert_attributes
```

Comment:
I guess collect_form_attributes seemed like the designated place to get "attributes to set on the object." My concern is that if we want those to be URIs and something else calls it, that it gets URIs.